docs/docs/tutorial/chapter3/saving-data.md

```diff
@@ -560,25 +638,24 @@ const CREATE_CONTACT = gql`
   }
 `
 
+interface FormValues {
+  name: string
+  email: string
+  message: string
+}
+
 const ContactPage = () => {
-  // highlight-start
-  const [create, { loading, error }] = useMutation(CREATE_CONTACT, {
-    onCompleted: () => {
-      toast.success('Thank you for your submission!')
-    },
-  })
-  // highlight-end
+  // highlight-next-line
+  const [create] = useMutation(CREATE_CONTACT)
 
-  const onSubmit = (data) => {
-    create({ variables: { input: data } })
+  const onSubmit: SubmitHandler<FormValues> = (data) => {
```

Comment:
```suggestion
  const onSubmit: SubmitHandler<FormValues> = (data: CreateContactInput) => {
```

Reply:
`data` is already typed thanks to `SubmitHandler<FormValues>`